pkg/ck8s/workload_cluster.go

```diff
@@ -230,11 +230,18 @@ func (w *Workload) requestJoinToken(ctx context.Context, name string, worker boo
 }
 
 func (w *Workload) RemoveMachineFromCluster(ctx context.Context, machine *clusterv1.Machine) error {
-	request := &apiv1.RemoveNodeRequest{Name: machine.Name, Force: true}
+	if machine == nil {
+		return fmt.Errorf("machine is nil")
+	} else if machine.Status.NodeRef == nil {
```

Comment:
nit: I'd split this into separate statements.
```
if condition1 {
  return ...
}
if condition2 {
  return ...
}
```
IMHO is slightly more readable.